File: pinyougou/pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * @Author: YangRunTao
 * @Description: 品牌服务接口
 * @Date: 2019/05/07 16:14
 * @Modified By:
 */
@SuppressWarnings("JavaDoc")
public interface BrandService {
    /**
     * @param
     * @description: 返回所有品牌
     * @return: java.util.List<com.pinyougou.pojo.TbBrand>
     * @author: YangRunTao
     * @date: 2019/05/07 16:15
     * @throws:
     **/
    List<TbBrand> findAll();

    /**
     * @param pageNum
     * @param pageSize
     * @description: 分页查询所有品牌
     * @return: PageResult<com.pinyougou.pojo.TbBrand>
     * @author: YangRunTao
     * @date: 2019/05/08 16:02
     * @throws:
     **/
    PageResult<TbBrand> findPage(Integer pageNum, Integer pageSize);

    /**
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @description: 条件分页查询所有品牌
     * @return: PageResult<com.pinyougou.pojo.TbBrand>
     * @author: YangRunTao
     * @date: 2019/05/09 11:18
     * @throws:
     **/
    PageResult<TbBrand> findPage(TbBrand tbBrand, Integer pageNum, Integer pageSize);

    /**
     * @param tbBrand
     * @description: 增加品牌
     * @return: void
     * @author: YangRunTao
     * @date: 2019/05/08 18:03
     * @throws:
     **/
    void addBrand(TbBrand tbBrand);

    /**
     * @param tbBrand
     * @description: 修改品牌
     * @return: void
     * @author: YangRunTao
     * @date: 2019/05/08 19:59
     * @throws:
     **/
    void updateBrand(TbBrand tbBrand);

    /**
     * @param id
     * @description: 根据id查询品牌
     * @return: com.pinyougou.pojo.TbBrand
     * @author: YangRunTao
     * @date: 2019/05/08 20:02
     * @throws:
     **/
    TbBrand findOneBrand(Long id);

    /**
     * @param ids
     * @description: 批量删除品牌
     * @return: Result
     * @author: YangRunTao
     * @date: 2019/05/09 9:34
     * @throws:
     **/
    void deleteBrand(Long[] ids);

    /**
     * @param
     * @description: 获得品牌数据
     * @return: java.util.List<java.util.Map>
     * @author: YangRunTao
     * @date: 2019/05/10 15:08
     * @throws:
     **/
    List<Map> selectOptionList();
}

File: pinyougou/pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
/**
 * 品牌服务层js
 */
app.service('brandService', function ($http) {
    /*查询所有品牌业务ajax*/
    this.findAll = function () {
        //获得后台的数据
        return $http.get('../brand/findAll.do');
    };

    /*分页ajax*/
    this.findPage = function (page, rows) {
        //获得后台的数据
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };

    /*增加品牌ajax*/
    this.addBrand = function (entity) {
        return $http.post('../brand/addBrand.do', entity);
    };

    /*修改品牌ajax*/
    this.updateBrand = function (entity) {
        return $http.post('../brand/updateBrand.do', entity);
    };

    /*查询指定品牌ajax*/
    this.findOneBrand = function (id) {
        return $http.get('../brand/findOneBrand.do?id=' + id);
    };

    /*批量删除品牌ajax*/
    this.deleteBrand = function (selectIds) {
        //获取选中的复选框
       return $http.get('../brand/deleteBrand.do?ids=' + selectIds);
    };

    /*条件分页查询ajax*/
    this.search = function (page, rows,searchEntity) {
       return $http.post('../brand/searchBrands.do?page=' + page + "&rows=" + rows,searchEntity);
    };

    //获得品牌id和品牌name
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});